File: API/src/accounts/dto/account_dto.rs
use rocket::figment::value::Num;
use rocket::serde::Serialize;

#[derive(Serialize)]
pub struct AccountDto {
    pub id: Num,
    pub code: String,
    pub description: String,
    pub currency: String,
    pub balance_cents: Num,
    pub balance_cents_in_ref_currency: Num,
    pub ref_currency: String,
    pub iban: String,
    pub status: String,
    #[serde(rename(serialize = "type"))]
    pub type_: String,
    pub can_create_transactions: String
}

File: Services/api_js/src/modules/accounts/account.dto.ts
/**
 * Data Transfer Object for Account responses
 */
export interface AccountDto {
  id: number;
  code: string;
  description: string;
  currency: string;
  iban: string;
  status: string;
  asset_type: string;
  pri_transactions_src: string;
  balance_cents: number;
  balance_eur_cents: number;
}
